Assets/Scripts/CartesianTiles/WFSGeoJSONTileDataLayer.cs

```diff
@@ -0,0 +1,144 @@
+        private string wfsUrl = "";
+        public string WfsUrl { get => wfsUrl; set => wfsUrl = value; }
+
+        private GeoJSONLayer geoJSONLayer;
+        public GeoJSONLayer GeoJSONLayer { 
+			get => geoJSONLayer; 
+			set
+			{
+				if(geoJSONLayer != null)
+					geoJSONLayer.ReferencedProxy.LayerDestroyed.RemoveListener(OnGeoJSONLayerDestroyed);
+
+				geoJSONLayer = value;
+				geoJSONLayer.ReferencedProxy.LayerDestroyed.AddListener(OnGeoJSONLayerDestroyed);
+			}
+		}
+
+        private void Awake() {
+			//Make sure Datasets at least has one item
+			if(Datasets.Count == 0)
+			{
+				var baseDataset = new DataSet()
+				{
+					maximumDistance = 3000,
+					maximumDistanceSquared = 1000 * 1000
+				};
+				Datasets.Add(baseDataset);
+			}
+
+			StartCoroutine(FindTileHandler());
+        }
+
+		private IEnumerator FindTileHandler()
+		{
+			yield return null;
+
+			//Find a required TileHandler in our parent, or else in the scene
+            tileHandler = GetComponentInParent<TileHandler>();
+            
+            if(!tileHandler)
+                tileHandler = FindAnyObjectByType<TileHandler>();
+
+            if(tileHandler){
+                tileHandler.AddLayer(this);
+                yield break;
+            }
+
+            Debug.LogError("No TileHandler found.", gameObject);
+		}
+
+		public override void HandleTile(TileChange tileChange, System.Action<TileChange> callback = null)
+		{
+			TileAction action = tileChange.action;
+			var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+			switch (action)
+			{
+				case TileAction.Create:
+					Tile newTile = CreateNewTile(tileKey);
+					tiles.Add(tileKey, newTile);
+					newTile.runningCoroutine = StartCoroutine(DownloadGeoJSON(tileChange, newTile, callback));
+					break;
+				case TileAction.Upgrade:
+					tiles[tileKey].unityLOD++;
+					break;
+				case TileAction.Downgrade:
+					tiles[tileKey].unityLOD--;
+					break;
+				case TileAction.Remove:
+					geoJSONLayer.RemoveFeaturesOutOfView();
+					
+					InteruptRunningProcesses(tileKey);
+					tiles.Remove(tileKey);
+					callback?.Invoke(tileChange);
+					return;
+				default:
+					break;
+			}
+		}
+
+        private void OnGeoJSONLayerDestroyed()
+        {
+            Destroy(gameObject);
+        }
+
+        private void OnDestroy() {
+            if(tileHandler)
+                tileHandler.RemoveLayer(this);
+        }
+
+		private Tile CreateNewTile(Vector2Int tileKey)
+		{
+            Tile tile = new()
+            {
+                unityLOD = 0,
+                tileKey = tileKey,
+                layer = transform.gameObject.GetComponent<Layer>()
+            };
+
+            return tile;
+		}
+
+		private IEnumerator DownloadGeoJSON(TileChange tileChange, Tile tile, System.Action<TileChange> callback = null)
+		{
+			var bboxValue = $"{tileChange.X},{tileChange.Y},{(tileChange.X + tileSize)},{(tileChange.Y + tileSize)}";
+			string url = WfsUrl.Replace("{0}", bboxValue);
```

Comment:
Zou je een comment bij de WfsUrl field kunnen plaatsen dat deze een placeholder `{0}` moet hebben waar de bbox ingezet wordt? Nu kom ik het tegen in de code. En het zou zelfs ook mooi zijn om in de onValidate en/of de setter een check hierop te hebben omdat dit verplicht is

Reply:
Yes, perfect.